src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldLikeEventSymbol.cs

```diff
@@ -239,6 +253,104 @@ internal SourceEventDefinitionAccessorSymbol(
             {
                 return null;
             }
+
+            protected override SourceMemberMethodSymbol? BoundAttributesSource
+            {
+                get
+                {
+                    return IsExtern && this.MethodKind == MethodKind.EventAdd
+                        ? (SourceMemberMethodSymbol?)this.AssociatedEvent.RemoveMethod
+                        : null;
+                }
+            }
+
+            protected override IAttributeTargetSymbol AttributeOwner
+            {
+                get
+                {
+                    Debug.Assert(PartialDefinitionPart is null);
```

Comment:
I suggest asserting `IsPartialDefinition` instead, as it helps remind us of the context we are in. It's not just "`this` better not be an implementation part", it's "`this` will *always* be a definition part, because it is a `SourceEventDefinitionAccessorSymbol`."

```suggestion
                    Debug.Assert(IsPartialDefinition);
```